tests/test_density.py

```diff
@@ -0,0 +1,170 @@
+from argparse import ArgumentParser
+import numpy as np
+import math
+from parcels import Field, Grid, JITParticle, ScipyParticle, Variable
+
+
+def GradientClimber(particle, grid, time, dt):
+    f_lat = dt / 1000. / 1.852 / 60.
+    f_lon = f_lat / math.cos(particle.lat*math.pi/180)
```

Comment:
OK, so the `mesh=flat` argument to the grid is just a short-hand notation for setting the appropriate `UnitConverter` classes on the `U` and `V` field. For an individual field you want to set the converter directly by passing `unit=GeographicPolar()`, which will make the conversion implicitly happen every time you sample the field `K` either from Python or within the kernels.